### ISSUE TITLE
Help with deploying own server very much appreciated

### ISSUE THREAD

**Issue by pjdevries** [beyondcode/expose#209]
First I like to thank Beyond Code for this cool project and making it available to the rest of us.

After many hours of tinkering, trying to get my own Expose server up, running and accessible, I'm now both frustrated and desperate. Hopefully I can get some help from community members who managed to make this work.

On my local development system, I use Docker. I have several projects, each with their own virtual host and accessible in my workstation's .local network. Exposing any of those via `sharedwithexpose.com` works flawlessly. Trying to expose them via my own Expose server is quite another story.

Let's assume I want to access the Expose server via subdomain `tunnel.myserver.tld` and port `18181`.
- A DNS `A` record was added for the subdomain. 
- A Let's Encrypt certificate was installed.
- The Nginx configuration for the virtual host was adjusted, reflecting the proposed configuration in the [SSL Support](https://beyondco.de/docs/expose/server/ssl) documentation. The essential parts now look like this:
```
server_name tunnel.myserver.tld *.tunnel.myserver.tld;

location / {
    proxy_pass http://127.0.0.1:18181; # Expose server
    proxy_read_timeout     60;
    proxy_connect_timeout  60;
    proxy_redirect         off;

    # Allow the use of websockets
    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection 'upgrade';
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Forwarded-Proto https;
    proxy_set_header Host $host;
    proxy_cache_bypass $http_upgrade;
}
```

I start the server using the following command:
`expose serve tunnel.myserver.tld --port=18181`

The client configuration looks like this:
```
return [
    'host' => 'tunnel.myserver.tld',
    'port' => 443,
    ...
    ...
    ...
];
```
I start the client using the following command:
`expose share foo.bar.local --subdomain=foo`

resulting in the following feed back from expose:
```
Thank you for using expose.
Local-URL:        foo.bar.local
Dashboard-URL:    http://127.0.0.1:4040
Expose-URL:       https://foo.tunnel.myserver.tld
```
Unfortunately `https://foo.tunnel.myserver.tld` does not take me to the website. It seems the address can not be resolved and I have no clue how to solve this. 

One thing in particular I don't quite understand, is the server configuration. The documentation has a special section for [Client Configuration](https://beyondco.de/docs/expose/client/configuration), explaining a default configuration file  `~/.expose/config.php` has to be generated first, which can be adjusted to ones needs. For the server there is a section in the documentation about the [Admin Interfac](https://beyondco.de/docs/expose/server/admin-interface) and [ Server Configuration](https://beyondco.de/docs/expose/server/server-configuration), but there is no mention of a configuration file and it's location. So where do I add the administrator credentials for instance?

A little help is very much appreciated.

**Comment by mpociot** [comment]
Hi @pjdevries ,

It looks like you are already halfway there. 
At least your expose client is able to connect to your custom server and authenticate with it.

Can you tell me how you started your Expose server?

Regarding the configuration, Expose is looking for the configuration file in a couple of locations.
If a `.expose.php` file exists within the working directory of the server, this will be used. Otherwise the server looks for a file in this location: `~/.expose/config.php`
See https://github.com/beyondcode/expose/blob/cca03ab8b290d975dd660ea66cf1b6f3b1e04e3e/app/Providers/AppServiceProvider.php#L48-L66

I think the problem comes from your DNS records.  Did you only add an A record for `tunnel.myserver.tld`?
You will need an A record for every subdomain too:

Make sure you have an A record for: `*.tunnel.myserver.tld`

**Comment by pjdevries** [comment]
Hi @mpociot,

Thanx for the reply and suggestions. 

The wildcard `A` record was already in place. Because I tried various scenario's, I found several config files, so I still have to figure out which one is being used at the moment. Because I could not think of anything else that might be wrong, I simply restarted server and client once more. Guess what? It works! I changed nothing whatsoever after my post here, but it works nevertheless.

I do run into issues with some of the local sites I try to expose, but I'll try to figure those out later. First clean things up and try to have a predictable, reproducible environment with a proper config file.

Thanks again so far.
